packages/adblocker-extended-selectors/test/unit/eval.test.ts

```diff
@@ -27,13 +27,9 @@ function testMatches(selector: string, html: string, target: string, expected: b
     return;
   }
 
-  const element = target === 'document' ? document : document.querySelector(target);
-  expect(element).to.not.be.null;
-  if (element !== null) {
-    // NOTE: here we need to ignore the type warnings so that we can pass a
-    // `Document` argument to test some edge cases (e.g. textContent returns
-    // null on document).
-    // @ts-ignore
+  const element = document.querySelector(target);
```

Comment:
Removing the `@ts-ignore` exposed a wrong assumption. `matches` is incapable of accepting `document` as it is not an Element. If needed it should be able to handle a `documentElement` which is a valid Element. Thanks to this realization we can remove two invalid tests.